Update WPK documentation.

DFolchA opened the pull request:
Hello team,

This PR closes https://github.com/wazuh/wazuh-packages/issues/196 and #1304 .

I have added documentation about how to use the tools within wazuh-packages to create a WPK automatically and updated the existing documentation to create a WPK for windows manually.

Regards
Daniel Folch

alberpilot commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+    -o,   --output <name>                       [Required] Name to the output package.
+    -h,   --help                                Show this help.
+
+3. Examples
```
3. Example title is smaller than 3.1.

alberpilot replied:
Title "examples". Why? Why not `wpk generation` or similar?

DFolchA replied:
The size of the titles has been corrected and I have changed the example title in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/f1780dd4ad9aa2688d10d80739e742caedb4139e

alberpilot commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+Automated custom WPK packages creation
+============================
+
+Wazuh provides an automated way of building WPK packages.
```
Here, a larger description is needed. We can mention what is the process for the WPK building: the use of docker, the output, the automation capability, etc.

DFolchA replied:
I have added a bit longer description in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/f1780dd4ad9aa2688d10d80739e742caedb4139e

alberpilot commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-installers/wpk
+
+2.2. Execute the `generate_wpk_package.sh` script, whith the diferent options you desire. This script will build a Docker image with all the necesary tools to  build the WPK and run a container that will build it:
```
whith?

alberpilot replied:
necesary?

DFolchA replied:
Typos corrected in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/5a5c8eee3bfe1d516cebc82e63143658667f151e

alberpilot commented:
I stopped to review, after the last "necesary" typo. I will continue with the review when the PR is free of typos.

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+2. Build WPK
+--------------------
+
+2.1. Download our wazyh-packages repository from github, there you can find tools for building diferent kinds of packages, and go to the wpk directory.
```
Typos are corrected in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/5a5c8eee3bfe1d516cebc82e63143658667f151e

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+
+Set the location as follows:
+
+    - /C=US is the country.
```
https://github.com/wazuh/wazuh-documentation/pull/1308/commits/3dec92570dfc246436fe31715f4687a965310bdf

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,106 @@
+.. _create-custom-wpk-automatically:
+
+Automated custom WPK packages creation
+============================
```
Done in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/f1780dd4ad9aa2688d10d80739e742caedb4139e

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+Wazuh provides an automated way of building WPK packages using docker so there is no need for any other dependency.
+
+To create a WPK package this way only docker and a X509 certificate are needed, in the case of Windows WPK a msi package is needed.
```
This sentence doesn't make sense. In addition, there are several typos: `docker`=> `Docker`, `msi` => `MSI`

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+To create a WPK package this way only docker and a X509 certificate are needed, in the case of Windows WPK a msi package is needed.
+
+1. Get a X509 certificate and CA
```
Remove the 1.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+1. Get a X509 certificate and CA
+--------------------------------
+
+1. Create root CA
```
This [README.md](https://github.com/wazuh/wazuh-packages/blob/master/wpk/README.md) from wazuh-packages repository has more information about the SSL certificate. Why isn't here?

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+1. Create root CA
+
+.. code-block:: console
```
The code-blocks from this section don't match the indentation.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+2. Build WPK
+-------------
+
+Requirements
```
This sub-sub section must be removed. You can write it as normal text. In addition, you are saying that you only need Docker, and 3 lines later you are using `git`.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+  .. code-block:: console
+
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-installers/wpk
```
wazuh-installers?

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-installers/wpk
+
+Execute the `generate_wpk_package.sh` script, with the different options you desire. This script will build a Docker image with all the necessary tools to create the WPK and run a container that will build it:
```
In RST language, the in-line code is marked using double ` at the beginning and at the end of the code.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+To use this tool the certificate and the key must be in the same directory.
+
+2.1. Linux WPK
```
Remove the 2.1. from the title.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+2.1. Linux WPK
+^^^^^^^^^^^^^^
+
+An example of how to build a Linux WPK would be:
```
I would change this line to: _Here you can see how to build a WPK package for Linux systems_.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+    # ./generate_wpk_package.sh -t linux -b branch_tag -d output/path -k path/to/keys -o myagent.wpk
+
+2.2. Windows WPK
```
Remove the 2.2.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+2.2. Windows WPK
+^^^^^^^^^^^^^^^^
+
+To build a WPK for Windows you need to first download a msi package of the desired version:
```
First, change msi to MSI.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+  .. code-block:: console
+
+    # curl -Ls https://packages.wazuh.com/3.x/windows/wazuh-agent-3.9.2-1.msi --output wazuh-agent-3.9.2-1.msi
```
You can change this curl command to `curl -O url`. This will download the package with the same name as it is in the upstream.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+
+    # curl -Ls https://packages.wazuh.com/3.x/windows/wazuh-agent-3.9.2-1.msi --output wazuh-agent-3.9.2-1.msi
+
+An example of how to build a Windows WPK package would be:
```
Change this to _here you can see how to build the WPK package for Windows_ or something like this.

DFolchA replied:
Done in d3a58a1, 473937b, ff99d41

BraulioV commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,102 @@
+    # /generate_wpk_package.sh -t windows -b branch_tag -d output/path -k path/to/keys -o myagent.wpk -pd path/to/wazuhagent.msi
+
+Definitions:
+    - **branch_tag** is the branch of the wazuh repository you want to build the sources from.
```
Change this to inline code.

DFolchA replied:
Done in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/d3a58a12dc60f26417cf452ebd4fb55d105aa227, https://github.com/wazuh/wazuh-documentation/pull/1308/commits/473937b3ad29f7b7880bf4b0a07f435b020c387b, https://github.com/wazuh/wazuh-documentation/pull/1308/commits/ff99d415dee956a8d309b5adae627a2677b8989f

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+Wazuh provides an automated way of building WPK packages using docker so there is no need for any other dependency.
+
+To create a WPK package your own WPK package follow these steps:
```
Done in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/1da3000084beaad7d5ea7d1da0a469b53c1a08be

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+Get an X509 certificate and CA
+--------------------------------
+
+If you already have your own certificate this step can be skipped.
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+            -c,   --checksum <path>                     [Optional] Generate checksum
+            -h,   --help                                Show this help.
+
+To use this tool the certificate and the key must be in the same directory.
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+Linux WPK
+----------
+
+Here you can see how to build a WPK package for Linux systems.
```
Done in 1da3000Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+ .. code-block:: console
+
+        # ./generate_wpk_package.sh -t linux -b branch_tag -d output/path -k path/to/keys -o myagent.wpk
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+        # curl -O https://packages.wazuh.com/3.x/windows/wazuh-agent-3.9.5-1.msi
+
+Here you can see how to build a WPK package for Windows systems.
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+ .. code-block:: console
+
+        # /generate_wpk_package.sh -t windows -b branch_tag -d output/path -k path/to/keys -o myagent.wpk -pd path/to/wazuhagent.msi
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+        # /generate_wpk_package.sh -t windows -b branch_tag -d output/path -k path/to/keys -o myagent.wpk -pd path/to/wazuhagent.msi
+
+If the -c or --checksum option is used there will be a file containing the SHA512 checksum in the same output path or you can indicate where you want to store it.
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,109 @@
+
+        # ./generate_wpk_package.sh -t linux -b branch_tag -d output/path -c path/to/checksum -k path/to/keys -o myagent.wpk
+
+Definitions:
```
Done in 1da3000

DFolchA commented on source/user-manual/agents/remote-upgrading/create-custom-wpk-manually.rst:
```diff
@@ -0,0 +1,175 @@
+Get an X509 certificate and CA
+--------------------------------
+
+Create root CA
```
Done in 1da3000

DFolchA commented:
Hello team,
I have changed the hierarchy of the files for the WPK generation and added a new file that contains how to create the key an certificate, this part was repeated twice before this.
Additionally, this file also contains links to the manual and automated method of generating the WPK.
Regards,
Daniel Folch

alberpilot commented on source/user-manual/agents/remote-upgrading/create-custom-wpk/create-custom-wpk-automatically.rst:
```diff
@@ -0,0 +1,75 @@
+            -c,   --checksum <path>                     [Optional] Generate checksum
+            -h,   --help                                Show this help.
+
+To use this tool the certificate and the key must be in the same directory.
```
Please, indicate here what certificate is needed. 
"To use this tool, the previously required certificated (link to the parent page) and the key must be in the same directory"

DFolchA replied:
Done in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/e89410a03a224177d448c2cb1b2aad358b372ed2
